### DISCUSSION TITLE
CAS Provider

### DISCUSSION THREAD

**Discussion by pmakowski23** [nextauthjs/next-auth#5632] [Help]
### Question 💬

Has anyone tried implementing NextAuth with CAS as the provider? I found an node cas client but I don't know how could I implement it as provider for NextAuth in my app. https://www.npmjs.com/package/node-cas-client

### How to reproduce ☕️

No reproduction needed.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by ThangHuuVu** [comment]
We have documentation to help create new provider here: https://next-auth.js.org/configuration/providers/oauth#adding-a-new-built-in-provider

Let me know if something is unclear 🙌

**Reply by jgribonvald** [reply]
The problem with CAS (like the SAML auth) is that use redirects instead of APIs endpoints like OAuth's protocols. 
Do you have other provider that works like that ? On an other way some peoples use the SAML-jackson like to "translate" SAML auth to OAuth flow.
